docs/api/asyncio_con.rst

```diff
@@ -234,116 +234,88 @@ Connection
+        than on the original pool itself.
 
-   async with connection.transaction():
-       await connection.execute(
-           'CREATE TYPE User { CREATE PROPERTY name -> str }')
+    .. py:method:: try_transaction()
 
-       try:
-           # Create a savepoint:
-           async with connection.transaction():
-               await connection.execute(
-                   "INSERT User { name := 'Don' }")
-               # This nested savepoint will be
-               # automatically rolled back:
-               raise Exception
-       except:
-           # Ignore exception
-           pass
+        Execute a non-retryable transaction.
 
-       # Because the nested savepoint was rolled back, there
-       # will be nothing in `User`.
-       assert (await connection.query('SELECT User')) == []
+        Contrary to ``retry()``, ``try_transaction()`` will not attempt
+        to re-run the nested code block in case a retryable error happens.
 
-Alternatively, transactions can be used without an ``async with`` block:
+        This is a low-level API and it is advised to use the ``retry()``
+        method instead.
 
-.. code-block:: python
-
-    tr = connection.transaction()
-    await tr.start()
-    try:
-        ...
-    except:
-        await tr.rollback()
-        raise
-    else:
-        await tr.commit()
+        A call to ``try_transaction()`` returns
+        :py:class:`AsyncIOTransaction`.
 
+        Example:
 
-See also the
-:py:meth:`AsyncIOConnection.transaction()` function.
+        .. code-block:: python
 
+            with con.try_transaction() as tx:
+                value = tx.query_one("SELECT Counter.value")
+                tx.execute(
+                    "UPDATE Counter SET { value := <int64>$value",
+                    value=value,
+                )
 
-.. py:class:: AsyncIOTransaction
+        Note executing queries on ``tx`` object rather than the original
+        connection.
 
-    Represents a transaction or savepoint block.
+    .. py:method:: transaction(isolation=None, readonly=None, deferrable=None)
+        **Deprecated**. Use :py:meth:`retry` or :py:meth:`try_transaction`.
 
-    Transactions are created by calling the
-    :py:meth:`AsyncIOConnection.transaction()` method.
+        Create a :py:class:`AsyncIOTransaction` object.
 
+        :param isolation:
+            Transaction isolation mode, can be one of:
+            ``'serializable'``, ``'repeatable_read'``.  If not specified,
+            the server-side default is used.
 
-    .. py:coroutinemethod:: start()
+        :param readonly:
+            Specifies whether or not this transaction is read-only.  If not
+            specified, the server-side default is used.
 
-        Enter the transaction or savepoint block.
+        :param deferrable:
+            Specifies whether or not this transaction is deferrable.  If not
+            specified, the server-side default is used.
```

Comment:
I'd also link our transaction docs here that explain what readonly or deferrable do.

Reply:
This is deprecated part. Let's add this to the `with_transaction_options`.